w3lib/http.py

```diff
@@ -90,10 +97,8 @@ def basic_auth_header(username, password, encoding="ISO-8859-1"):
 
     """
 
-    auth = "%s:%s" % (username, password)
-    if not isinstance(auth, bytes):
-        # XXX: RFC 2617 doesn't define encoding, but ISO-8859-1
-        # seems to be the most widely used encoding here. See also:
-        # http://greenbytes.de/tech/webdav/draft-ietf-httpauth-basicauth-enc-latest.html
-        auth = auth.encode(encoding)
-    return b"Basic " + urlsafe_b64encode(auth)
+    auth = "%s:%s" % (to_native_str(username), to_native_str(password))
+    # XXX: RFC 2617 doesn't define encoding, but ISO-8859-1
```

Comment:
is this comment still needed? I think it was about the `encode` call which is no longer done?

Reply:
I think it applies as long as we accept an `encoding` parameter here with that default value. We are still using the `encoding`, only now it has moved to the `to_bytes` call below.